lib/Sema/CSSimplify.cpp

```diff
@@ -9880,6 +9880,11 @@ bool ConstraintSystem::resolveClosure(TypeVariableType *typeVar,
     }
   }
 
+  SolutionApplicationTarget target(closure, closure,
+                                   ContextualTypePurpose::CTP_CallArgument,
+                                   contextualType, /*isDiscarded=*/false);
+  setSolutionApplicationTarget(closure, target);
```

Comment:
@xedin I’m pretty sure this part is not correct. As we discussed, I’m using `SolutionApplicationTarget` to convey the contextual type of the `ClosureExpr`. I’m not entirely sure what you’d expect for the decl context, contextual type purpose and `isDiscarded`. AFAICT, we don’t really know the contextual type purpose and whether it’s discarded inside `resolveClosure`.

Reply:
Thanks. I changed it.